### PR TITLE
lightining,expression: support user variable for `BuildSimpleExpr` and remove `PlanContext` dependency in lightning context

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: ref #53388

### What changed and how does it work?

- `BuildSimpleExpr` support build accessing user variables like `@a + 1`
- Remove planContext dependency in lightning `Session` context to make it simple.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by D3Hunter — COMMENTED** [review]
5/9 reviewd

File: pkg/planner/core/expression_rewriter.go [hunk 1]
```diff
@@ -1651,37 +1660,54 @@ func (er *expressionRewriter) useCache() bool {
...
+
+	sessionVars, err := contextopt.SessionVarsPropReader{}.GetSessionVars(evalCtx)
+	if err != nil {
+		er.err = err
+		return
+	}
+
+	if er.planCtx != nil {
+		intest.Assert(sessionVars == er.planCtx.builder.ctx.GetSessionVars())
+	}
```
Comment (D3Hunter) [inline pkg/planner/core/expression_rewriter.go:1680]:
```suggestion
	intest.Assert(er.planCtx == nil || sessionVars == er.planCtx.builder.ctx.GetSessionVars())
```

**Review by D3Hunter — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Comment by yudongusa** [comment]
What about testing results, any regression or new issues?

**Pull request merged**